app/models/grouping.rb

```diff
@@ -740,6 +740,16 @@ def final_grade(result)
   end
 
   # Helper for populate_submissions_table.
+  # Returns the total bonus/deductions for this grouping other than late penalty.
+  def total_extra_points(result)
+    if !result.nil? && result.marking_state == Result::MARKING_STATES[:complete]
```

Comment:
Two comments here.

1. This seems like it belongs in the `Result` model, not `Grouping`.
2. Let's always show the points, even if the result is not marked as complete. This way, late penalties will show immediately (once they're added).